### DISCUSSION TITLE
add timeout when visit disk and adjust ResponseHeaderTimeout(15min) in http client config?

### DISCUSSION THREAD

**Discussion by daydreamjf** [minio/minio#16963] [Q&A] [answered]
when disk is slow, request should wait util the slowest disk return, that may cause client timeout and too many requests wait to be handled. so should we adjust  config or add something here ?

**Comment by klauspost** [comment, marked as answer]
If you system cannot handle the load you can adjust the S3 API request limit. `mc admin config set ALIAS api requests_max=N`

**Reply by daydreamjf** [reply]
thanks for your answer, yes, adjust api request limit can adjust cluster load。 but disk slow is not the only way to be caused. for example, disk is in some error (i have met a io hang sense) or  set load is not balanced or else。 i think here we should use a more effective timeout mechanism， and why here is 15miniutes?

**Reply by klauspost** [reply]
Yes. Generally S3 requests should not have timeouts.

**Reply by daydreamjf** [reply]
Clients or server should not have timeouts? In my opinion， when we access external dependencies， we should have timeouts. Expect more explanations and discussions

**Reply by klauspost** [reply]
@daydreamjf If you want explanations and discussions look into our [subscriptions](https://min.io/pricing).

The short version: Uploads and downloads are often gigabytes of data so timeouts are quite disruptive.
